syncer/syncer.go

```diff
@@ -357,13 +362,11 @@ func (s *Syncer) setStateAfterSync(ctx context.Context, success bool) {
 	}
 }
 
-func (s *Syncer) syncLayer(ctx context.Context, layerID types.LayerID) (*types.Layer, error) {
+func (s *Syncer) syncLayer(ctx context.Context, layerID types.LayerID) (layer *types.Layer, err error) {
```

Comment:
i am not a fan of named returned values. i've seen quite a bit of programming mistakes that assign the values by accident. it's not very obvious for debugging. i see you return explicitly but it leaves room for ppl who edit the code later. i really think the extra two lines of var declaration is worth it.